x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/rules_settings_modal.tsx

```diff
@@ -192,69 +123,90 @@ export const RulesSettingsModal = memo((props: RulesSettingsModalProps) => {
+    setting: keyof RulesSettingsProperties,
+    key: keyof RulesSettingsFlappingProperties | keyof RulesSettingsQueryDelayProperties,
+    value: boolean | number
   ) => {
-    if (!settings) {
-      return;
+    if (setting === 'flapping') {
+      if (!flappingSettings) {
+        return;
+      }
+      const newSettings = {
+        ...flappingSettings,
+        [key]: value,
+      };
+      setFlappingSettings({
+        ...newSettings,
+        statusChangeThreshold: Math.min(
+          newSettings.lookBackWindow,
+          newSettings.statusChangeThreshold
+        ),
+      });
+      setHasFlappingChanged(true);
     }
 
-    const newSettings = {
-      ...settings,
-      [key]: value,
-    };
-
-    setSettings({
-      ...newSettings,
-      statusChangeThreshold: Math.min(
-        newSettings.lookBackWindow,
-        newSettings.statusChangeThreshold
-      ),
-    });
+    if (setting === 'queryDelay') {
+      if (!queryDelaySettings) {
+        return;
+      }
+      const newSettings = {
+        ...queryDelaySettings,
+        [key]: value,
+      };
+      setQueryDelaySettings(newSettings);
+      setHasQueryDelayChanged(true);
+    }
   };
 
   const handleSave = () => {
-    if (!settings) {
-      return;
+    const updatedSettings: RulesSettingsProperties = {};
+    if (canWriteFlappingSettings && hasFlappingChanged) {
+      updatedSettings.flapping = flappingSettings;
+    }
+    if (canWriteQueryDelaySettings && hasQueryDelayChanged) {
+      updatedSettings.queryDelay = queryDelaySettings;
     }
-    mutate(settings);
+    mutate(updatedSettings);
   };
 
   if (!isVisible) {
     return null;
   }
 
   const maybeRenderForm = () => {
-    if (hasError || !canShowFlappingSettings) {
+    if (!canShowFlappingSettings && !canShowQueryDelaySettings) {
       return <RulesSettingsErrorPrompt />;
     }
-    if (!settings || isLoading) {
+    if (isFlappingLoading || isQueryDelayLoading) {
       return <CenterJustifiedSpinner />;
     }
     return (
-      <EuiForm>
-        <EuiFlexGroup>
-          <EuiFlexItem>
-            <RulesSettingsFlappingTitle />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-        <EuiSpacer size="s" />
-        <EuiFlexGroup>
-          <RulesSettingsModalFormLeft
-            isSwitchDisabled={!canWriteFlappingSettings}
-            settings={settings}
-            onChange={(e) => handleSettingsChange('enabled', e.target.checked)}
+      <>
+        {flappingSettings && (
+          <RulesSettingsFlappingSection
+            onChange={(key, value) => handleSettingsChange('flapping', key, value)}
+            settings={flappingSettings}
+            canWrite={canWriteFlappingSettings}
+            canShow={canShowFlappingSettings}
+            hasError={hasFlappingError}
           />
-          <RulesSettingsModalFormRight
-            settings={settings}
-            onChange={(key, value) => handleSettingsChange(key, value)}
+        )}
+        {queryDelaySettings && (
```

Comment:
Need a little padding between the sections when flapping is disabled?

<img width="877" alt="Screenshot 2023-10-12 at 9 14 46 AM" src="https://github.com/elastic/kibana/assets/13104637/f066dac3-2ec2-4907-842e-164f5f7f6a67">

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/167433/commits/e32a0a8d8d2b7e5d5770f153996463a806002456